howl/client/howl_client.py

```diff
@@ -75,19 +75,20 @@ def _on_audio(self, in_data, frame_count, time_info, status):
 
         # Inference from input sequence
         if self.engine.infer(inp):
-            # Check if inference window has passed to prevent callbacks from executing repeatedly
-            cur_time = time.time()
-            time_from_inference_ms = (cur_time - self._last_inference_time) * 1000
-            if time_from_inference_ms < self.engine.max_window_size_ms:
+            # Check if inference has already occured for this sequence to prevent
+            # duplicate callback execution
+            if self._infer_detected:
                 return data_ok
 
-            self._last_inference_time = time.time()
+            self._infer_detected = True
             phrase = ' '.join(self.ctx.vocab[x]
                               for x in self.engine.sequence).title()
             logging.info(f'{phrase} detected')
             # Execute user-provided listener callbacks
             for lis in self.listeners:
                 lis(self.engine.sequence)
+        else:
+            self._infer_detected = False
```

Comment:
How was the detection after this change. were you seeing duplicated detection?

Reply:
No more duplicate detection with this change 👍